<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>WidgetEditor</class>
 <widget class="QWidget" name="WidgetEditor">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1002</width>
    <height>807</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>0</width>
    <height>0</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QGridLayout" name="gridLayout" columnstretch="1,0">
   <item row="3" column="1">
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="1" column="1">
    <widget class="QFrame" name="actionFrame">
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Sunken</enum>
     </property>
     <layout class="QGridLayout" name="gridLayout_2" columnstretch="1,1">
      <item row="10" column="0" colspan="2">
       <widget class="QPushButton" name="square_button">
        <property name="text">
         <string>Square</string>
        </property>
       </widget>
      </item>
      <item row="28" column="0" colspan="2">
       <widget class="QPushButton" name="save_button">
        <property name="text">
         <string>Save</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QSpinBox" name="led_count">
        <property name="maximum">
         <number>1024</number>
        </property>
       </widget>
      </item>
      <item row="8" column="1">
       <widget class="QPushButton" name="v_line_button">
        <property name="text">
         <string>V-Line</string>
        </property>
       </widget>
      </item>
      <item row="6" column="0">
       <widget class="QPushButton" name="h_flip_button">
        <property name="text">
         <string>H-Flip</string>
        </property>
       </widget>
      </item>
      <item row="27" column="1">
       <widget class="QPushButton" name="cancel_button">
        <property name="text">
         <string>Cancel</string>
        </property>
       </widget>
      </item>
      <item row="14" column="1">
       <widget class="QPushButton" name="shrink_button">
        <property name="text">
         <string>Shrink</string>
        </property>
       </widget>
      </item>
      <item row="27" column="0">
       <widget class="QPushButton" name="reset_button">
        <property name="text">
         <string>Reset</string>
        </property>
       </widget>
      </item>
      <item row="6" column="1">
       <widget class="QPushButton" name="v_flip_button">
        <property name="text">
         <string>V-Flip</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="width">
        <property name="text">
         <string>Width</string>
        </property>
       </widget>
      </item>
      <item row="17" column="0" colspan="2">
       <widget class="QCheckBox" name="auto_identify">
        <property name="text">
         <string>Auto identify</string>
        </property>
       </widget>
      </item>
      <item row="12" column="0" colspan="2">
       <widget class="QPushButton" name="zigzag_button">
        <property name="text">
         <string>ZigZag</string>
        </property>
       </widget>
      </item>
      <item row="14" column="0">
       <widget class="QPushButton" name="grow_button">
        <property name="text">
         <string>Grow</string>
        </property>
       </widget>
      </item>
      <item row="13" column="0" colspan="2">
       <widget class="QPushButton" name="sawtooth_button">
        <property name="text">
         <string>Sawtooth</string>
        </property>
       </widget>
      </item>
      <item row="5" column="0" colspan="2">
       <widget class="QPushButton" name="rotate_button">
        <property name="text">
         <string>Rotate</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QSpinBox" name="w_spinBox">
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>1024</number>
        </property>
       </widget>
      </item>
      <item row="15" column="0" colspan="2">
       <widget class="QPushButton" name="auto_resize_button">
        <property name="text">
         <string>Auto resize</string>
        </property>
       </widget>
      </item>
      <item row="4" column="0" colspan="2">
       <widget class="QPushButton" name="undo_button">
        <property name="text">
         <string>Undo</string>
        </property>
       </widget>
      </item>
      <item row="18" column="0" colspan="2">
       <widget class="QPushButton" name="identify_button">
        <property name="text">
         <string>Identify</string>
        </property>
       </widget>
      </item>
      <item row="8" column="0">
       <widget class="QPushButton" name="h_line_button">
        <property name="text">
         <string>H-Line</string>
        </property>
       </widget>
      </item>
      <item row="11" column="0" colspan="2">
       <widget class="QPushButton" name="circle_button">
        <property name="text">
         <string>Circle</string>
        </property>
       </widget>
      </item>
      <item row="26" column="0" colspan="2">
       <widget class="QPushButton" name="copy_shape_button">
        <property name="text">
         <string>Copy shape</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QSpinBox" name="h_spinBox">
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>1024</number>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="height">
        <property name="text">
         <string>Height</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="leds_count">
        <property name="text">
         <string>LEDS</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="1" column="0" rowspan="3">
    <widget class="QFrame" name="gridFrame">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>0</height>
      </size>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QGridLayout" name="gridLayout_3">
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <property name="spacing">
       <number>0</number>
      </property>
      <item row="0" column="0">
       <widget class="EditorGrid" name="grid">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>500</width>
          <height>100</height>
         </size>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>EditorGrid</class>
   <extends>QGraphicsView</extends>
   <header location="global">EditorGrid.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>w_spinBox</tabstop>
  <tabstop>h_spinBox</tabstop>
  <tabstop>led_count</tabstop>
  <tabstop>undo_button</tabstop>
  <tabstop>rotate_button</tabstop>
  <tabstop>h_flip_button</tabstop>
  <tabstop>v_flip_button</tabstop>
  <tabstop>h_line_button</tabstop>
  <tabstop>v_line_button</tabstop>
  <tabstop>square_button</tabstop>
  <tabstop>circle_button</tabstop>
  <tabstop>zigzag_button</tabstop>
  <tabstop>sawtooth_button</tabstop>
  <tabstop>grow_button</tabstop>
  <tabstop>shrink_button</tabstop>
  <tabstop>auto_resize_button</tabstop>
  <tabstop>auto_identify</tabstop>
  <tabstop>identify_button</tabstop>
  <tabstop>copy_shape_button</tabstop>
  <tabstop>reset_button</tabstop>
  <tabstop>cancel_button</tabstop>
  <tabstop>save_button</tabstop>
  <tabstop>grid</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
